### PR TITLE
Move ContainterPort to config and only for bazel jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
So that hopefully we can run multiple pods per node and save prow from weekend?

cc @Random-Liu

### CODE REVIEW

File: prow/plank/controller.go [hunk 1]
```diff
@@ -347,14 +347,6 @@ func (c *Controller) startPod(pj kube.ProwJob) (string, string, error) {
 	for i := range pj.Spec.PodSpec.Containers {
 		spec.Containers = append(spec.Containers, pj.Spec.PodSpec.Containers[i])
 		spec.Containers[i].Name = fmt.Sprintf("%s-%d", podName, i)
-		// Set the HostPort to 9999 for all build pods so that they are forced
```
Comment (ixdy) [inline prow/plank/controller.go]: maybe retain the comment somewhere?

Comment (krzyzacy) [inline prow/plank/controller.go]: I'm adding a test, and probably leave this comment there

**Comment by rmmh** [comment]
There are a bunch more jobs here that use .cache that you missed.

**Comment by krzyzacy** [comment]
awwww I thought only 'bazel' jobs...

anything mount `/root/.cache` or anything actually uses `--git-cache` flag?

**Comment by rmmh** [comment]
both, really

**Comment by krzyzacy** [comment]
what... then it helps nothing :-(

or those ci-e2e jobs does not actually need to mount cache-ssd?

**Comment by rmmh** [comment]
They use the cache to essentially retain a .git directory between invocations. A local mirror is a more effective way to achieve the same result.

**Comment by krzyzacy** [comment]
hummmm most of them don't check out anything though... maybe only for those use --repo flags?

**Comment by krzyzacy** [comment]
I'll remove all cache volume mount from ci-e2e jobs, since they are not using `--git-cache` flag when calling bootstrap.py. (They do check out test-infra, but that's the first thing even before bootstrap), and this can allow us to schedule multiple periodic e2e pods on the same machine, thoughts?

**Comment by fejta** [comment]
/lgtm

My hypothesis is that this will work but we should only deploy this change while we are around to debug.

**Comment by krzyzacy** [comment]
muhahahaha now that I'm oncall so *($^(#&(

gonna try tomorrow when I back to my seat

**Comment by krzyzacy** [comment]
I'll merge this and see what's gonna happen

**Pull request merged**